cfe_internal/enterprise/federation/federation.cf

```diff
@@ -154,11 +171,12 @@ bundle agent federation_manage_files
     enabled::
       "$(cfengine_enterprise_federation:config.bin_dir)/."
         create => "true",
-        perms => default:mog( "600", "root", "root" );
+        perms => default:mog( "660", "root", "$(transport_user)" );
     am_superhub::
       "$(cfengine_enterprise_federation:config.federation_dir)/superhub/."
         create => "true",
-        perms => default:mog( "701", "root", "root" );
+# TODO why did this have 701??? change and test.
+        perms => default:mog( "770", "root", "$(transport_user)" );
```

Comment:
It's fine to have this as 77, it's explicitly what you want, just remember that by default, any time you set read bit, you get execute bit. So, above for bin_dir, 660, will actually be 770. This may seem super weird, but it's super duper useful when you apply for permissions recursively. With one promise, you can usually get what you want on both files and directory instead of having to do recurseive dir promise, then seperately a recursive file promise (file_selct)

Reply:
@nickanderson yes,but IMO this should be explicit - `rxdirs` should default to false ;)

Reply:
Maybe, but for as long as I have used cfengine it has defaulted to the "do what I mean" behavior. It would be very disruptive to change that default now. We can introduce new bodies where rxdirs is not on by default, default:mog_explicit or some other better name.